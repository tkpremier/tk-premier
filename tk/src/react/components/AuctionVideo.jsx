import React from 'react';
import PropTypes from 'prop-types';

const AuctionVideo = props => (
  <div className="auction-video">
    <div className="caption row col-xs-12 clearfix">
      <div className="left col-xs-6">
        <p className="eyebrow"><em>{props.liveAuctionSuperTitle}</em></p>
        <h2>{props.liveAuctionTitle}</h2>
      </div>
      <div className="right col-xs-6">
        <a className="video-link" href={props.liveAuctionLinkUrl}>{props.liveAuctionLinkDesc}</a>
      </div>
    </div>
    <div className="video-wrapper">
      <iframe
        src={props.liveAuctionUrl}
        allowFullScreen
        title={props.liveAuctionLinkDesc}
      />
    </div>
  </div>
);

AuctionVideo.propTypes = {
  liveAuctionTitle: PropTypes.string,
  liveAuctionSuperTitle: PropTypes.string,
  liveAuctionLinkUrl: PropTypes.string,
  liveAuctionLinkDesc: PropTypes.string,
  liveAuctionUrl: PropTypes.string
};


export default AuctionVideo;
