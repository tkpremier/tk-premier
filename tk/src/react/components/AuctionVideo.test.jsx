import React from 'react';
import AuctionVideo from './AuctionVideo';
import renderer from 'react-test-renderer';

it('should render an auction video', () => {
    const auctionvideo = renderer.create(
        <AuctionVideo
            liveAuctionSuperTitle="will be inside p className eyebrow>"
            liveAuctionTitle="will be the h2 tag"
            liveAuctionLinkUrl="will be inside link className video-link
            liveAuctionLinkDesc="will be inside link className video-link
            liveAuctionUrl="will be the src of the iframe"
        />
    );
    const tree = auctionvideo.toJSON();
    expect(tree).toMatchSnapshot();
});
