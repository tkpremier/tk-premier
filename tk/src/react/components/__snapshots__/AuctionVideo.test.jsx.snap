// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an auction video 1`] = `
<div
  className="auction-video"
>
  <div
    className="caption row col-xs-12 clearfix"
  >
    <div
      className="left col-xs-6"
    >
      <p
        className="eyebrow"
      >
        <em>
          will be inside p className eyebrow&gt;
        </em>
      </p>
      <h2>
        will be the h2 tag
      </h2>
    </div>
    <div
      className="right col-xs-6"
    >
      <a
        className="video-link"
        href="will be inside link className video-link
 liveAuctionLinkDesc="
      />
    </div>
  </div>
  <div
    className="video-wrapper"
  >
    <iframe
      allowfullscreen={true}
      mozallowfullscreen={true}
      src="will be the src of the iframe"
      webkitallowfullscreen={true}
    />
  </div>
</div>
`;
